From f882d0347261a6e245e8983930801711f4dc6a57 Mon Sep 17 00:00:00 2001
From: Jiri Moskovcak <jmoskovc@redhat.com>
Date: Thu, 12 Jul 2012 13:01:15 +0200
Subject: [PATCH 082/105] abrt-hook-ccpp: save "chrooted" /etc/system-release
 too (if exists)

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>

Conflicts:
	src/daemon/abrt-server.c
	src/plugins/abrt-dump-xorg.c (deleted)
---
 src/daemon/abrt-server.c     | 3 ++-
 src/hooks/abrt-hook-ccpp.c   | 7 ++++---
 src/plugins/abrt-dump-oops.c | 2 +-
 3 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/daemon/abrt-server.c b/src/daemon/abrt-server.c
index 15f3370..dc8329e 100644
--- a/src/daemon/abrt-server.c
+++ b/src/daemon/abrt-server.c
@@ -113,7 +113,8 @@ static int create_debug_dump()
     {
         error_msg_and_die("Error creating crash dump %s", path);
     }
-    dd_create_basic_files(dd, client_uid);
+    dd_create_basic_files(dd, client_uid, NULL);
+    dd_save_text(dd, "abrt_version", VERSION);
 
     dd_save_text(dd, FILENAME_ANALYZER, analyzer);
     dd_save_text(dd, FILENAME_EXECUTABLE, executable);
diff --git a/src/hooks/abrt-hook-ccpp.c b/src/hooks/abrt-hook-ccpp.c
index 1a918c8..6f9c57b 100644
--- a/src/hooks/abrt-hook-ccpp.c
+++ b/src/hooks/abrt-hook-ccpp.c
@@ -690,7 +690,9 @@ int main(int argc, char** argv)
     dd = dd_create(path, fsuid, 0640);
     if (dd)
     {
-        dd_create_basic_files(dd, fsuid);
+        char *rootdir = get_rootdir(pid);
+
+        dd_create_basic_files(dd, fsuid, (rootdir && strcmp(rootdir, "/") != 0) ? rootdir : NULL);
 
         char source_filename[sizeof("/proc/%lu/somewhat_long_name") + sizeof(long)*3];
         int source_base_ofs = sprintf(source_filename, "/proc/%lu/smaps", (long)pid);
@@ -729,12 +731,10 @@ int main(int argc, char** argv)
         dd_save_text(dd, FILENAME_PID, pid_str);
         if (user_pwd)
             dd_save_text(dd, FILENAME_PWD, user_pwd);
-        char *rootdir = get_rootdir(pid);
         if (rootdir)
         {
             if (strcmp(rootdir, "/") != 0)
                 dd_save_text(dd, FILENAME_ROOTDIR, rootdir);
-            free(rootdir);
         }
 
         char *reason = xasprintf("Process %s was killed by signal %s (SIG%s)",
@@ -855,6 +855,7 @@ int main(int argc, char** argv)
             trim_problem_dirs(g_settings_dump_location, maxsize * (double)(1024*1024), path);
         }
 
+        free(rootdir);
         return 0;
     }
 
diff --git a/src/plugins/abrt-dump-oops.c b/src/plugins/abrt-dump-oops.c
index 8a24458..85ca2c0 100644
--- a/src/plugins/abrt-dump-oops.c
+++ b/src/plugins/abrt-dump-oops.c
@@ -154,7 +154,7 @@ static unsigned save_oops_to_dump_dir(GList *oops_list, unsigned oops_cnt)
 
         if (dd)
         {
-            dd_create_basic_files(dd, /*uid:*/ my_euid);
+            dd_create_basic_files(dd, /*uid:*/ my_euid, NULL);
             dd_save_text(dd, "abrt_version", VERSION);
             dd_save_text(dd, FILENAME_ANALYZER, "Kerneloops");
             dd_save_text(dd, FILENAME_KERNEL, first_line);
-- 
1.7.11.2

